### PR TITLE
Image of a function

### PR DESCRIPTION
**Pull request by felixwellen**
Construct the image, show that an embedding, (co-)restricted to its image is an equivalence.

(see this external issue:
https://github.com/felixwellen/synthetic-geometry/issues/12)

### CODE REVIEW

File: Cubical/Functions/Image.agda [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+private variable
+  ℓ ℓ' : Level
+  A B : Type ℓ
+
+module _ (f : A → B) where
+  isInImage : B → Type _
+  isInImage y = ∃[ x ∈ A ] f x ≡ y
+
+  image : Type _
```
Comment (mortberg) [inline Cubical/Functions/Image.agda:28]: Shouldn't this be `Image` as it's a type?

File: Cubical/Functions/Image.agda [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+open import Cubical.HITs.PropositionalTruncation as PT
+open import Cubical.Data.Sigma
+
+private variable
+  ℓ ℓ' : Level
+  A B : Type ℓ
+
+module _ (f : A → B) where
+  isInImage : B → Type _
+  isInImage y = ∃[ x ∈ A ] f x ≡ y
```
Comment (mortberg) [inline Cubical/Functions/Image.agda:26]: For completeness please add the trivial lemma that this is a prop

File: Cubical/Functions/Image.agda [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+module Cubical.Functions.Image where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Equiv.Fiberwise
+open import Cubical.Foundations.Univalence
```
Comment (mortberg) [inline Cubical/Functions/Image.agda:11]: A lot of imports for a small file. Do you really need all of them?

**Review by mortberg — CHANGES_REQUESTED** [review]
A few tiny comments, then good to merge!

File: Cubical/Functions/Image.agda (hunk 1 above)
Comment (felixwellen) [reply Cubical/Functions/Image.agda]: changed

File: Cubical/Functions/Image.agda (hunk 2 above)
Comment (felixwellen) [reply Cubical/Functions/Image.agda:23]: Done

File: Cubical/Functions/Image.agda (hunk 3 above)
Comment (felixwellen) [reply Cubical/Functions/Image.agda]: Your suspicion was correct - not all of them were needed (anymore..).

**Review by mortberg — APPROVED** [review]

**Comment by mortberg** [comment]
Great! Merging once the CI is finished

**Pull request merged**